db/version_set.cc

```diff
@@ -4892,6 +4892,13 @@ Status VersionSet::ProcessManifestWrites(
       assert(manifest_io_status.ok());
     }
     if (s.ok() && new_descriptor_log) {
+      Directories directories_;
+      if (db_directory == nullptr) {
```

Comment:
Why would db_directory be nullptr in the depths of ProcessManifestWrites()? Feels like callers might not be passing db_directory properly.

Reply:
Fixed by requiring such nullptr in upper callers.